src/Cache/Internal/ScsDataClient.php

```diff
@@ -1093,6 +1102,87 @@ function () use ($call): SetRemoveElementResponse {
         );
     }
 
+    /**
+     * @return ResponseFuture<GetBatchResponse>
+     */
+    public function getBatch(string $cacheName, array $keys): ResponseFuture
+    {
+        try {
+            validateCacheName($cacheName);
+            validateKeys($keys);
+
+            $getRequests = [];
+            foreach ($keys as $key) {
+                $getRequest = new _GetRequest();
+                $getRequest->setCacheKey($key);
+                $getRequests[] = $getRequest;
+            }
+
+            $getBatchRequest = new _GetBatchRequest();
+            $getBatchRequest->setItems($getRequests);
+            $call = $this->grpcManager->client->GetBatch($getBatchRequest, ['cache' => [$cacheName]], ['timeout' => $this->timeout]);
+        } catch (SdkError $e) {
+            return ResponseFuture::createResolved(new GetBatchError($e));
+        } catch (Exception $e) {
+            return ResponseFuture::createResolved(new GetBatchError(new UnknownError($e->getMessage(), 0, $e)));
+        }
+
+        return ResponseFuture::createPending(
+            function () use ($call): GetBatchResponse {
+                try {
+                    $responses = $call->responses();
```

Comment:
we really don't have to do anything special in order to aggregate / loop over the streaming responses? This function just does all of that?

Reply:
The $calls->responses return the generator function. We iterate over the generator function inside the Success call and create our final result and values

Reply:
Okay, I think I get that. Can you add some tests for the Async variants too then? I'd like to see what it looks like to consume them so I can reason about when the loop happens relative to the `wait`.

Reply:
It looks like we're missing a call to `$call->getStatus()` to make sure the call was actually successful. I'm able to pass a nonexistent cache name to both `getBatch` and `setBatch` and they appear to succeed.

Reply:
We cannot wait after the call has been created since the responses are still streaming. I can create a status function in the success call to see what was the call status after the wait was performed on it?

Reply:
Re: `wait`: okay; maybe I just need to see how a user would consume our future then.

Reply:
and maybe i'm talking about the `wait` on our `Future` class, not one from grpc

Reply:
I am doing a wait inside the getBtach function in CacheClient: https://github.com/momentohq/client-sdk-php/pull/177/files#diff-faee61621cf99402304d5ff4af286157f07f3fb59e14fefcd672fd0a90a79e84R1219. 
Is this what you're looking for?

Reply:
yeah but I would still like to see what it looks like for an end user who wants to use Async, and I think it's worth having some basic test coverage for those Asyncs